engine/example_test.go

```diff
@@ -64,7 +64,7 @@ func Example_advertiseHelloWorld() {
 
 	// Note that this example publishes an ad with bitswap metadata as an example.
 	// But it does not instantiate a bitswap server to serve retrievals.
-	adCid, err := engine.NotifyPut(context.Background(), []byte(sayHelloCtxID), md)
+	adCid, err := engine.NotifyPut(context.Background(), "", []string{}, []byte(sayHelloCtxID), md)
```

Comment:
`nil` suffices as empty slice. In go you can use `nil` for empty slice and calling `len` on it will return zero.

Ditto in other places.